### PR TITLE
Update detection of guest clients

### PR DESCRIPTION
**Pull request by sfan5**

### CODE REVIEW

**Comment by nerzhul** [comment]
note sure it's a good enhancement this could match nrz255 for example

**Comment by sfan5** [comment]
It won't, it only matches if the first letter if uppercase.

File: server.py [hunk 1]
```diff
@@ -288,12 +288,10 @@ def sort(self):
...
 			points = 0
 
-			# 1 per client, but only 1/8 per client with a guest
-			# or all-numeric name.
+			# 1 per client, but only 1/8 per "guest" client
 			if "clients_list" in server:
 				for name in server["clients_list"]:
-					if name.startswith("Guest") or \
-							name.isdigit():
+					if re.match(r"[A-Z][a-z]+[0-9]{3,4}", name):
```
Comment (ShadowNinja) [inline server.py]: Should be pre-compiled.

Comment (sfan5) [inline server.py]: The docs say:

> Note: The compiled versions of the most recent patterns passed to re.compile() and the module-level matching functions are cached, so programs that use only a few regular expressions at a time needn’t worry about compiling regular expressions.

-- https://docs.python.org/3/library/re.html#re.compile

**Comment by est31** [comment]
So when I read that correctly, players with the name Celeron55 will still be counted, as they only have two digits?

**Comment by sfan5** [comment]
No, only player with:
* first char is uppercase letter
* all following chars are lowercase letters
* 3 or 4 numbers at the end

**Comment by est31** [comment]
Nice

**Comment by sfan5** [comment]
(I already have this applied in production anyway)

**Pull request merged**